source/server-discovery-and-monitoring/server-monitoring.rst

```diff
@@ -700,7 +709,6 @@ The event API here is assumed to be like the standard `Python Event
             return new ServerDescription
         except (NetworkError, CommandError) as exc:
             close connection
-            connection = Null
```

Comment:
Removed the assignment to avoid needing a lock here. Closing the connection is enough as it will be re-created on the next check. This also makes the code easier to reason about imo because `connection` is only null if it's the first check and closed if the previous check was cancelled.

Reply:
Acknowledged.